src/platform/magento2/o2m.js

```diff
@@ -27,301 +12,41 @@ function isNumeric (val) {
-                })
-              }).catch(err => {
-                logger.error('Error placing an order', err, typeof err)
-                if (job) job.attempts(6).backoff({delay: 30 * 1000, type: 'fixed'}).save()
-                return done(composeError('Error placing an order.', err));
-              })
-            }).catch((errors) => {
-              logger.error('Error while adding shipping address', errors)
-              if (job) job.attempts(3).backoff({ delay: 60 * 1000, type: 'fixed' }).save()
-              return done(composeError('Error while adding shipping address.', errors));
-            })
-          }).catch((errors) => {
-            logger.error('Error while adding billing address', errors)
-            if (job) job.attempts(3).backoff({ delay: 60 * 1000, type: 'fixed' }).save()
-            return done(composeError('Error while adding billing address.', errors));
-          })
-        }).catch((errors) => {
-          logger.error('Error while synchronizing country list', errors)
-          if (job) job.attempts(3).backoff({ delay: 30 * 1000, type: 'fixed' }).save()
-          return done(composeError('Error while syncing country list.', errors));
-        })
-      }).catch((errors) => {
-        logger.error('Error while adding products', errors)
-        if (job) job.attempts(3).backoff({ delay: 30 * 1000, type: 'fixed' }).save()
-        return done(composeError('Error while adding products.', errors));
-      })
-    })
-  }
-
-  cartIdPrepare.then(processCart).catch((error) => { // cannot create a quote for specific user, so bypass by placing anonymous order
+  try {
+    cartId = await orderProcessor.cartIdPrepare(orderData);
+    orderData.cart_id = cartId;
+  } catch (error) { // cannot create a quote for specific user, so bypass by placing anonymous order
     logger.error(THREAD_ID, error)
     logger.info('< Bypassing to anonymous order')
-    isThisAuthOrder = false
 
     if (isNumeric(cartId)) { // we have numeric id - assigned to the user provided
-      api.cart.create(null, null).then((result) => {
-        processCart(result)
-        //      logger.info('< Assigning guest cart with the user')
-        //      api.cart.assign(cartId, userId).then((subres) =>{
-        //        console.info(subres)
-        //        processCart(result)
-        //      }).catch((err) => {
-        //        logger.error(err)
-        //      })
-      }).catch(error => {
-        logger.info(error)
-        return done(composeError('Error while adding products.', error));
-      }) // TODO: assign the guest cart with user at last?
-    } else {
-      logger.info(THREAD_ID + '< Using cartId provided with the order', cartId)
-      processCart(cartId)
+      cartId = await apiUtilities.getApi(orderData.store_code).cart.create(null, null);
+      orderData.cart_id = cartId;
     }
-  })
+  }
+
+  orderProcessor.processCart({ job, orderData, done });
```

Comment:
Please `await` it. In `order.js` someone might try to do something after processSingleOrder